packages/bump-monorepo-packages/src/bump-packages.js

```diff
@@ -1,14 +1,18 @@
+#!/usr/bin/env node
```

Comment:
I can't comment on the unchanged lines, but if we are planning to use this in compass, we will need to change this script

> // we increment the package version based on the bump on dependencies:
> // if a devDependency was bumped, regardless of the increment we increment of a
> // patch, otherwise we replicate the increment of the dependency. We always keep
> // the biggest increase.

Dependencies that are bundled with the package will be a dev dependency (it doesn't need to be explicitly installed when package is used, but it is part of distribution) and so assuming that the version bump can be a patch will be incorrect

Reply:
If I understand it correctly, if, for example, we change `compass-components` and add a new feature there, that would also be a new feature where is used, but `compass-components` is a development dependency of other packages and so the dependants would not be bumped as a feature bump, right?

I'm happy to find a solution for this, though I'm not too concerned, it would be rare that the work for a breaking change in a "library" package won't need changes on dependants too, that means you'd have the same "breaking" or "feat" commit also on the dependants which would bump them accordingly.

What would be a good alternative that doesn't need us to enter each version manually? Should we have the option to ignore the "type" of dependency? That sounds a bit drastic, for example, a breaking change in the eslint config would cause a major bump everywhere.

Reply:
I wonder .. is there a way to detect those packages that are meant to be used as webpack? or worse case we can come up with something like a custom prop? what do you think?

Reply:
In compass we parse the source to detect what are the production dependencies of the package: https://github.com/mongodb-js/compass/blob/a43c2b20c67310ba845d88709a3e1b0339c421a2/scripts/check-peer-deps.js#L9-L42

I guess we can adapt the same thing here, this script does expect all the source to be in the `src` folder though, which we might need to adapt somehow, maybe use src by default but make it changeable through some package.json option or something

Reply:
Another thing here that I see is that this script is missing actual tagging and commiting part. Lerna doesn't support just running `version` to produce a commit with tags so if we are planning to use this script, we would need to move this logic here (or in an additional script) too

Reply:
> Another thing here that I see is that this script is missing actual tagging and commiting part.

Yeah, that would be intentional, I'd literally do this: 
- https://github.com/mongodb-js/devtools-shared/blob/main/.github/workflows/bump-packages.yaml

And:

- https://github.com/mongodb-js/devtools-shared/blob/main/.github/workflows/publish-packages.yaml

No need to tag, unless we really want to, just to have a pointer to the commit where a bump happened, i don't think that's a requirement, we only did that cause lerna did that.

Reply:
I found tags super helpful to find and compare changes between releases, so ideally if it's not much effort I would try to keep them. This script seem to already have all the required info that you'd need to correctly tag a release and commit

Reply:
That would be fine, not critical, yeah. Just to clarify the flow I'm using sometimes is to just search for the tag in GitHub UI and then use the compare feature right there:

![image](https://user-images.githubusercontent.com/5036933/180473708-559700ea-b317-4478-a778-4aae55342aaf.png)